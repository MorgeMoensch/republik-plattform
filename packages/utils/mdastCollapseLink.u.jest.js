const UUT = require('./mdastCollapseLink')

const node = {
  type: 'root',
  children: [
    {
      type: 'paragraph',
      children: [
        {
          type: 'text',
          value: 'Kurzer Link: ',
        },
        {
          type: 'link',
          title: null,
          url: 'https://www.ise.fraunhofer.de',
          children: [
            {
              type: 'text',
              value: 'https://www.ise.fraunhofer.de',
            },
          ],
        },
      ],
    },
    {
      type: 'paragraph',
      children: [
        {
          type: 'text',
          value: 'Langer Link: ',
        },
        {
          type: 'link',
          title: null,
          url: 'https://www.ise.fraunhofer.de/content/dam/ise/en/documents/publications/studies/Stromerzeugung_2017_e.pdf',
          children: [
            {
              type: 'text',
              value:
                'https://www.ise.fraunhofer.de/content/dam/ise/en/documents/publications/studies/Stromerzeugung_2017_e.pdf',
            },
          ],
        },
      ],
    },
    {
      type: 'paragraph',
      children: [
        {
          type: 'link',
          title: null,
          url: 'https://www.ise.fraunhofer.de/content/dam/ise/en/documents/publications/studies/Stromerzeugung_2017_e.pdf',
          children: [
            {
              type: 'text',
              value: 'Beschrifteter Link',
            },
          ],
        },
      ],
    },
    {
      type: 'paragraph',
      children: [
        {
          type: 'text',
          value: 'Textpassage',
        },
      ],
    },
  ],
}

describe('mdastCollapseLink', () => {
  test('collapse link nodes', () => {
    const result = UUT(node)
    expect(result).toMatchSnapshot()
  })

  test('collapse link nodes w/ options', () => {
    const result = UUT(node, {
      prefixLength: 10,
      postfixLength: 5,
      collapseChar: 'â€“',
    })
    expect(result).toMatchSnapshot()
  })
})
