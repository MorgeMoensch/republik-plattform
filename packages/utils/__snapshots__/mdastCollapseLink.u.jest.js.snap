// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdastCollapseLink collapse link nodes 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Kurzer Link: ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "https://www.ise.fraunhofer.de",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://www.ise.fraunhofer.de",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Langer Link: ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "https://www.ise.fraunhofer.de/conte…2017_e.pdf",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://www.ise.fraunhofer.de/content/dam/ise/en/documents/publications/studies/Stromerzeugung_2017_e.pdf",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Beschrifteter Link",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://www.ise.fraunhofer.de/content/dam/ise/en/documents/publications/studies/Stromerzeugung_2017_e.pdf",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Textpassage",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Hack",
                },
              ],
              "type": "emphasis",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://de.wikipedia.org/wiki/Hack",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`mdastCollapseLink collapse link nodes w/ options 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Kurzer Link: ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "https://ww–er.de",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://www.ise.fraunhofer.de",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Langer Link: ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "https://ww–e.pdf",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://www.ise.fraunhofer.de/content/dam/ise/en/documents/publications/studies/Stromerzeugung_2017_e.pdf",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Beschrifteter Link",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://www.ise.fraunhofer.de/content/dam/ise/en/documents/publications/studies/Stromerzeugung_2017_e.pdf",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Textpassage",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Hack",
                },
              ],
              "type": "emphasis",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://de.wikipedia.org/wiki/Hack",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;
